sdk/messaging/azservicebus/messageSettler.go

```diff
@@ -181,7 +201,33 @@ func (s *messageSettler) DeadLetterMessage(ctx context.Context, message *Receive
 			}
 		}
 
-		if s.useManagementLink(message, receiver) {
+		var err error
+
+		if shouldSettleOnReceiver(message) {
+			ms.notifySettleOnLink(message)
+
+			info := map[string]any{
+				"DeadLetterReason":           reason,
+				"DeadLetterErrorDescription": description,
+			}
+
+			if options != nil && options.PropertiesToModify != nil {
+				for key, val := range options.PropertiesToModify {
+					info[key] = val
+				}
+			}
+
+			amqpErr := amqp.Error{
+				Condition: "com.microsoft:dead-letter",
+				Info:      info,
+			}
+
+			err = receiver.RejectMessage(ctx, message.RawAMQPMessage.inner, &amqpErr)
+		}
+
+		if shouldSettleOnMgmtLink(err, message) {
```

Comment:
Same fall-through question.

Reply:
I added a comment in all four locations.